### DISCUSSION TITLE
How can the PyCharm plugin guide V2 migration?

### DISCUSSION THREAD

**Discussion by koxudaxi** [pydantic/pydantic#5344] [Pydantic V2]
@samuelcolvin @dmontagu 

I'm working on Pydantic V2 support for the PyCharm plugin.
https://github.com/koxudaxi/pydantic-pycharm-plugin/issues/694
I plan for the plugin to show the warning for the deprecated thing and the link to the migration document in the warning popup.
Also, The user will be able to run a migration script from the popup.

Do you have the other idea to support the migration?

The plugin was 10k unique downloads only last month.
If the plugin can guide the migration and help users for v2, it avoids many questions for migration and v2 in pydantic issues and discussion from users.

I created the issue on the plugin project. But, I would share the topic with many people. And I post it here.

**Comment by samuelcolvin** [comment]
Thanks so much @koxudaxi, I'll reply on https://github.com/koxudaxi/pydantic-pycharm-plugin/issues/694.
